src/main/java/net/barrage/tegridy/validation/validator/Ip4Validator.java

```diff
@@ -0,0 +1,41 @@
+package net.barrage.tegridy.validation.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import net.barrage.tegridy.validation.annotation.Ip4;
+
+/** Validates whether the given string field is a valid IPv4 address. */
+public class Ip4Validator implements ConstraintValidator<Ip4, Object> {
+  static final String IP4_REGEX = "^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\\b){4}$";
```

Comment:
Do we want to allows notation with subnets mask, or maybe create separate validation for ips with subnet mask?

Reply:
Definitely separate